rpcs3/rpcs3qt/vfs_dialog_tab.cpp

```diff
@@ -83,3 +94,13 @@ void vfs_dialog_tab::AddNewDirectory()
 
 	m_dirList->setCurrentItem(new QListWidgetItem(dir, m_dirList));
 }
+
+void vfs_dialog_tab::RemoveDirectory()
+{
+	int row = m_dirList->currentRow();
+	if (row == 0) // First row is always a default directory, don't remove it
```

Comment:
hazard! currentRow can and will return -1 when no item is selected

Reply:
I hear you! You think `row > 0` as success condition is safe, then?

Reply:
Just check <=

Reply:
We were also wondering about those random changes you threw in there

Reply:
I replaced equality check with `<=` now.

About those "random changes", they are outlined in the first point of the PR - uses move semantics for shared pointers so they don't end up being copied. Should I split those into a separate commit of this PR? IMO it'd make sense, as then one commit would be about move semantics, while the other one would be solely about changed visuals of the dialog.

Reply:
Could you explain the const ref removal?

Reply:
It allows to utilize move semantics for `vfs_settings_info`, and since it's a non-POD style, it'll save a copy.

Consider the code flow before my change:
1) `vfs_settings_info` gets **constructed**. Constructor for `vfs_dialog_tab` references it via a const reference, so there's no copy done.
2) `info` gets assigned to `m_info`. `info` is a const reference, and thus cannot be modified, thus this involves a **copy**.

Result: 1 construction, 1 copy.

Now after my changed:
1) `vfs_settings_info` gets **constructed**. Constructor for `vfs_dialog_tab` references it via value, so the constructed value is used directly (at least in this case - notice this parameter is always constructed in place, when constructing `vfs_dialog_tab`).
2) `info` gets assigned to `m_info`. `info` is a value, and thus **can** be modified. Therefore, `std::move(info)` will **move** `info` to `m_info`.

Result: 1 construction, 1 move.

A rule of thumb I always stick to in such cases is - if the function takes ownership of an object (let it be a constructor or any normal function) **or** wants to mutate it, pass the argument by value to allow the callee to utilize move semantics properly. In cases where the value is not mutated or ownership is not taken from the callee, const reference works best.

Reply:
weird. I guess it's fine..